File: src/server/BuildingBlock/MovieTicket.Domain/Entities/Movie.cs
using MovieTicket.Core.Domain;

namespace MovieTicket.Domain.Entities;

public class Movie : EntityBase
{
    public string Name { get; set; }
    public DateTime ReleaseDate { get; set; }
    public long TotalTime { get; set; }
    public string Description { get; set; }
    
    public string Avatar { get; set; }
    public string Trailer { get; set; }
    
    public virtual ICollection<Category> Categories { get; set; }
    
    public virtual ICollection<Screening> Screenings { get; set; }
    
}

File: src/client/mobile/app/src/main/java/com/superman/movieticket/domain/entities/Movie.kt
package com.superman.movieticket.domain.entities

import androidx.datastore.core.Serializer
import com.superman.movieticket.core.entity.BaseEntity
import java.io.Serializable
import java.math.BigInteger
import java.util.Date

data class Movie (
    var name: String,
    var releaseDate: String,
    var totalTime: Number,
    var description: String,
    var avatar: String,
    var trailer: String,
    var categories: List<Category>
): BaseEntity(),Serializable

